warehouse/transformer/safe.go

```diff
@@ -0,0 +1,273 @@
+		snakeCaseFn = snakecase.ToSnakeCaseWithNumbers
+	}
+
+	name := strings.Join(extractAlphanumericValues(tableName), "_")
+	if strings.HasPrefix(tableName, "_") {
+		name = reLeadingUnderscores.FindString(tableName) + snakeCaseFn(reLeadingUnderscores.ReplaceAllString(name, ""))
+	} else {
+		name = snakeCaseFn(name)
+	}
+	if startsWithDigit(name) {
+		name = "_" + name
+	}
+	return name
+}
+
+func transformColumnNameCached(tec *transformEventContext, key string) string {
+	var columnName string
+
+	cacheKey := transformColumnNameCacheKey{
+		destType:                tec.event.Metadata.DestinationType,
+		columnName:              key,
+		underscoreDivideNumbers: tec.destOpts.underscoreDivideNumbers,
+		useBlendoCasing:         tec.intrOpts.useBlendoCasing,
+	}
+	columnNameCached, ok := tec.cache.transformColumnNameCache.Load(cacheKey)
+	if !ok {
+		columnName = transformColumnName(tec.event.Metadata.DestinationType, tec.intrOpts, tec.destOpts, key)
+		tec.cache.transformColumnNameCache.Store(cacheKey, columnName)
+	} else {
+		columnName = columnNameCached.(string)
+	}
+	return columnName
+}
+
+// transformColumnName applies transformation to the input column name based on the destination type and configuration options.
+// If `useBlendoCasing` is enabled, it transforms the column name into Blendo casing.
+// Otherwise, it applies a more general transformation using the `transformName` function.
+func transformColumnName(destType string, intrOpts *intrOptions, destOpts *destOptions, columnName string) string {
+	if intrOpts.useBlendoCasing {
+		return transformNameToBlendoCase(destType, columnName)
+	}
+
+	var snakeCaseFn func(s string) string
+	if destOpts.underscoreDivideNumbers {
+		snakeCaseFn = snakecase.ToSnakeCase
+	} else {
+		snakeCaseFn = snakecase.ToSnakeCaseWithNumbers
+	}
+
+	name := strings.Join(extractAlphanumericValues(columnName), "_")
+	if strings.HasPrefix(columnName, "_") {
+		name = reLeadingUnderscores.FindString(columnName) + snakeCaseFn(reLeadingUnderscores.ReplaceAllString(name, ""))
+	} else {
+		name = snakeCaseFn(name)
+	}
+	if startsWithDigit(name) {
+		name = "_" + name
+	}
+	if destType == whutils.POSTGRES {
+		name = misc.TruncateStr(name, 63)
+	}
+	return name
+}
+
+func startsWithDigit(name string) bool {
+	if len(name) > 0 && (rune(name[0]) >= '0' && rune(name[0]) <= '9') {
+		return true
+	}
+	return false
+}
+
+// transformNameToBlendoCase converts the input string into Blendo case format by replacing non-alphanumeric characters with underscores.
+// If the name does not start with a letter or underscore, it adds a leading underscore.
+// The name is truncated to 63 characters for Postgres, and the result is converted to lowercase.
+func transformNameToBlendoCase(destType, name string) string {
+	key := reNonAlphanumericOrDollar.ReplaceAllString(name, "_")
+
+	if !reStartsWithLetterOrUnderscore.MatchString(key) {
+		key = "_" + key
+	}
+	if destType == whutils.POSTGRES {
+		key = misc.TruncateStr(name, 63)
```

Comment:
63 is getting repeated at multiple places

Reply:
Added a constant.